openedx/core/djangoapps/safe_sessions/tests/test_utils.py

```diff
@@ -118,14 +130,20 @@ def assert_request_user_mismatch(self, user_at_request, user_at_response):
             yield
 
     @contextmanager
-    def assert_session_user_mismatch(self, user_at_request, user_in_session):
-        """
-        Asserts that the logger was called when request.user at request
-        time doesn't match the request.user at response time.
-        """
-        with self.assert_logged(
-            "SafeCookieData user at request '{}' does not match user in session: '{}'".format(
-                user_at_request, user_in_session
-            ),
+    def assert_request_and_session_user_mismatch(self, user_at_request, user):
```

Comment:
What was the reason to combine the individual test methods?
Per Clean Code's "Single Responsibility" principle, and for readability, why not have each method assert a single thing?  Introducing either "if, or, and, but" in the docstring and in the method name should have a very good reason.